### PR TITLE
Add index name to ICreateIndexResponse

### LINKED ISSUE #3823: CreateIndex doesn't return index name
**NEST/Elasticsearch.Net version**:
6.8.0

**Elasticsearch version**:
6.8.0

**Describe the feature**:
Currently `CreateIndexAsync` returns an `ICreateIndexResponse` which includes the `ShardsAcknowledged` and `Acknowledged` properties but not the index name as returned in the json response:

```
{
  "acknowledged" : true,
  "shards_acknowledged" : true,
  "index" : "my-index-2019.06.17-000001"
}
```

This is useful when indices are created using Date Math patterns like `%3Cmy-index-%7Bnow%2Fd%7D-000001%3E` so the date doesn't has to be calculated again.

### PR DESCRIPTION
**Pull request by codebrain**
Fixes #3823

Port to `7.x` requires attention to new serialiser attributes.

### CODE REVIEW

File: src/Tests/Tests/Indices/IndexManagement/CreateIndex/CreateIndexApiTests.cs [hunk 1]
```diff
@@ -206,6 +206,9 @@ protected override void ExpectResponse(ICreateIndexResponse response)
 			response.Acknowledged.Should().BeTrue();
 			response.ShardsAcknowledged.Should().BeTrue();
 
+			if (Cluster.ClusterConfiguration.Version >= "6.8.0")
```
Comment (russcam) [inline src/Tests/Tests/Indices/IndexManagement/CreateIndex/CreateIndexApiTests.cs]: index is returned in Elasticsearch 6.0.0, so this condition is not needed

Comment (codebrain) [reply src/Tests/Tests/Indices/IndexManagement/CreateIndex/CreateIndexApiTests.cs]: ++ will remove

**Comment by russcam** [comment]
@codebrain Does this need to be ported to 7.x?

**Comment by codebrain** [comment]
@russcam Yes. Done.

**Pull request merged**